drivers/xen/events.c

```diff
@@ -21,6 +21,80 @@ static evtchn_handle_t event_channels[EVTCHN_2L_NR_CHANNELS];
 
 static void empty_callback(void *data) { }
 
+int evtchn_alloc_unbound(domid_t dom, domid_t remote_dom)
+{
+	int rc;
+	struct evtchn_alloc_unbound alloc = {
+		.dom = dom,
+		.remote_dom = remote_dom,
+	};
+
+	rc = HYPERVISOR_event_channel_op(EVTCHNOP_alloc_unbound, &alloc);
+	if (rc == 0) {
+		rc = alloc.port;
+	}
+
+	return rc;
+}
+
+int evtchn_bind_interdomain(domid_t remote_dom, evtchn_port_t remote_port)
+{
+	int rc;
+	struct evtchn_bind_interdomain bind = {
+		.remote_dom = remote_dom,
+		.remote_port = remote_port,
+	};
+
+	rc = HYPERVISOR_event_channel_op(EVTCHNOP_bind_interdomain, &bind);
+	if (rc == 0) {
+		rc = bind.local_port;
+	}
+
+	return rc;
+}
+
+int evtchn_status(evtchn_status_t *status)
+{
+	return HYPERVISOR_event_channel_op(EVTCHNOP_status, status);
+}
+
+int evtchn_unmask(evtchn_port_t port)
```

Comment:
Probably I wrongly understood the purpose of this hypercall. I checked how Linux deals with unmasking. Now I see how it should be used and what `unmask_event_channel()` is doing wrong.

I will fixed it in another commit/PR (both hypercall unmasking and usage of `sys_bitfield_*`), because Zephyr haven't atomic helper bitops for ARM (like sync_set_bit on Linux), but both problems needs this. I already faced with this problem previously, but it was possible to fix it with compiler built-in atomics.